infra/providers.tf

```diff
@@ -17,15 +17,15 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 3.25.0"
+      version = "~> 5.4"
     }
     helm = {
       source  = "hashicorp/helm"
-      version = "~> 2.9"
+      version = "~> 2.11"
     }
     kubernetes = {
       source  = "hashicorp/kubernetes"
-      version = "2.23.0"
+      version = "~> 2.23"
```

Comment:
The `~>` (squiggly arrow) of the version constraint only allows bumps to the right-most number — so this will allow `2.23.2` and `2.24` but not `3.0`. We're just avoiding bumps to the major version.